deploifai/api/__init__.py

```diff
@@ -410,3 +409,213 @@ def create_project(self, project_name: str, cloud_profile: CloudProfile, fragmen
             raise DeploifaiAPIError("Could not create project. Please try again.")
         except KeyError:
             raise DeploifaiAPIError("Could not create project. Please try again.")
+
+    def get_training_infrastructure_plans(self, uses_gpu: bool, cloud_provider: CloudProfile):
+        query = (
+            """
+            query($usesGpu: Boolean $whereProvider: CloudProvider){
+            trainingInfrastructurePlans(usesGpu: $usesGpu provider: $whereProvider){
+                plan
+                config{
+                  ... on AWSFalconDefaultConfig{
+                    ec2InstanceType
+                  }
+                  ... on AzureFalconDefaultConfig{
+                    vmSize
+                  }
+                  ... on GCPFalconDefaultConfig{
+                    computeMachineType
+                  }
+                }
+              }
+            }
+            """
+        )
+        variables = {
+            "usesGpu": uses_gpu,
+            "whereProvider": cloud_provider
+        }
+
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": query, "variables": variables},
+                headers=self.headers,
+            )
+            api_data = r.json()
+
+            return api_data["data"]["trainingInfrastructurePlans"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+
+    def falcon_ml_config(self, variable):
+        query = (
+            """    
+            query ($where: FalconMLConfigWhereInput)
+                {
+                    falconMLConfigs(where: $where){
+                        id
+                        pythonVersion
+                        frameworkVersion
+                        cudaVersion
+                        cudnnVersion
+                    }
+                }
+            """
+        )
+        variables = {
+            "where": variable
+        }
+
+        try:
+            r = requests.post(
+                self.uri,
+                json={"query": query, "variables": variables},
+                headers=self.headers,
+            )
+            api_data = r.json()
+
+            return api_data["data"]["falconMLConfigs"]
+        except TypeError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
+        except KeyError:
+            raise DeploifaiAPIError("Could not get information. Please try again.")
```

Comment:
Should be clearer with the exception messages, try `Could not get ML configs ...`  instead of just saying `... information`

Reply:
done